packages/api/src/controllers/room.ts

```diff
@@ -64,14 +72,163 @@ function toExternalRoom(room: Room, isAdmin = false) {
   if (isAdmin) {
     return room;
   }
-  room.userId = undefined;
-  for (const key in room.participants) {
-    room.participants[key].tracksPublished = undefined;
+  const roomForResp = {
+    id: room.id,
+    updatedAt: room.updatedAt,
+    createdAt: room.createdAt,
+    participants: room.participants,
+  };
+
+  for (const key in roomForResp.participants) {
+    roomForResp.participants[key].tracksPublished = undefined;
   }
-  room.events = undefined;
-  return room;
+  return roomForResp;
 }
 
+app.delete("/:roomId", authorizer({}), async (req, res) => {
+  const room = await db.room.get(req.params.roomId);
+
+  if (!req.user.admin && req.user.id !== room.userId) {
+    throw new ForbiddenError(`users may only delete their own rooms`);
+  }
+
+  const svc = new RoomServiceClient(
+    req.config.livekitHost,
+    req.config.livekitApiKey,
+    req.config.livekitSecret
+  );
+  await svc.deleteRoom(req.params.roomId);
+
+  room.deleted = true;
+  room.deletedAt = new Date().getTime();
+  await db.room.replace(room);
```

Comment:
I think an isolated update might be cleaner/safer in general than full replaces (and maybe even cheaper, given this room object can get pretty big). i.e. wdytof
```suggestion
  await db.room.update(room.id, { deleted: true, deletedAt: Date.now() });
```

Reply:
oh nice yeah didn't realise we could do that 👍